Update the ReadMe

Vyzer9 opened the pull request:
N/A – Moderate documentation improvement, no issues with Jira.

### Tests completed

N/A – Changes only to Markdown. No functional code affected.

### Proposed changelog entries

- Improve README.md with a better structure, usage instructions, and a table of contents to improve the visuals and efficiency of the readme documentation, guidelines for contributors, and clarity in explanations with more content and even photos showing the user what the site looks like.

### Proposed changelog category

/label skip-changelog

### Proposed update guidelines

N/A

<img width="1102" height="888" alt="Captura de tela de 2025-10-02 20-54-04" src="https://github.com/user-attachments/assets/32ee18fe-b6bf-470d-8766-1db78bf41891" />

<img width="1102" height="888" alt="Captura de tela de 2025-10-02 20-54-15" src="https://github.com/user-attachments/assets/7b52a2f6-7fd2-4710-b49e-c740f38190a4" />

<img width="1102" height="888" alt="Captura de tela de 2025-10-02 20-54-19" src="https://github.com/user-attachments/assets/a583c9eb-94bc-4cd9-b7bb-dd3315242e30" />

Commit:
Update README.md

Vyzer9 commented:
Hello, I've noticed the lack of content in the readme for some time now. Even though it's not that important, it helps a lot, and I've made the adjustments. If you need more content, if something is wrong or broken, or if I need to change the position, please let me know. Thank you, and I'm happy to help the Jenkins community. ❤️

MarkEWaite reviewed (commented):
Thanks for your interest in contributing to Jenkins.  Unfortunately, there are many errors in your proposed changes.

The pull request would have benefited significantly if you had checked that the commands  work that are included in the pull request.

This looks like it might have been generated by AI without sufficient proofreading by the user submitting the text.  Please spend your time as a contributor to assure that the pull request is accurate.  There are many more contributors than there are reviewers.

MarkEWaite reviewed (changes requested):
The formatting of the file needs to be corrected as well.  Instructions for formatting are included in the [contributing guide](https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#linting).

Commit:
Update README.md

Vyzer9 commented on README.md:
```diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
 [![Jenkins Regular Release](https://img.shields.io/endpoint?url=https%3A%2F%2Fwww.jenkins.io%2Fchangelog%2Fbadge.json)](https://www.jenkins.io/changelog)
 [![Jenkins LTS Release](https://img.shields.io/endpoint?url=https%3A%2F%2Fwww.jenkins.io%2Fchangelog-stable%2Fbadge.json)](https://www.jenkins.io/changelog-stable)
 
+# Getting Started (Development)
```
I understand your feedback, I removed the part with the incorrect step by step and just put the message with the link that directs to the part of the website where there is documentation with the steps

Vyzer9 commented on README.md:
````diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+### 2. Build the Project
+Use Maven to compile and package Jenkins:
+```bash
+mvn clean install -DskipTests
````
done

Vyzer9 commented on README.md:
````diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+- Logs will be shown in the console to help with debugging.
+- You can access the initial admin password in:
+```bash
+~/.jenkins/secrets/initialAdminPassword
````
I also confess that it was my mistake, I removed that part

Vyzer9 commented on README.md:
````diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+```bash
+MAVEN_OPTS="-Xmx2g -XX:MaxPermSize=512m"
+```
+- Permission issues on Unix/macOS: use sudo or ensure correct ownership of files.
````
okay ❤️

Vyzer9 commented on README.md:
````diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+### 5. Troubleshooting
+OutOfMemoryError: Try increasing memory limits:
+```bash
+MAVEN_OPTS="-Xmx2g -XX:MaxPermSize=512m"
````
removed :)

Vyzer9 commented on README.md:
```diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+| Run with tests      | `mvn clean install`             |
+| Start in dev mode   | `mvn hpi:run`                   |
+| Run unit tests only | `mvn test`                      |
+| Build WAR only      | `mvn package -DskipTests`       |
```
I agree, thank you for saying that, I've already removed it

Vyzer9 commented on README.md:
```diff
@@ -39,36 +58,148 @@ For all distributions Jenkins offers two release lines:
+| Build WAR only      | `mvn package -DskipTests`       |
+
+
+For more information on setting up your development environment, contributing, and working with Jenkins internals, check the official Jenkins Developer Documentation:
+➡️ [https://www.jenkins.io/doc/developer/](https://www.jenkins.io/doc/developer/)
```
okay

Commit:
Update README.md

Vyzer9 commented:
@MarkEWaite I've already resolved all the change requests, which I fully agree with. I tested the links, fixed them, and removed anything deemed unnecessary and repetitive by the user. During testing, we encountered errors, but in my view, these were related to my PR and not to the Jenkins environment, since I didn't change any code other than the ReadMe.

If you have any other requests, please let me know so I can change them, or even if you want to make them more user-friendly and lively, for example.

Anyway, I'm immensely grateful for your help, even though I made mistakes and fixed them.Thanks 👍🏻 ❤️

MarkEWaite committed:
Remove link to non-existent newsletter

Also format with `yarn lint:fix` as noted in the contributing guide.

MarkEWaite commented:
In the future, please use a branch that is named something other than `master`.  When you use the branch named `master`, you complicate your local work with the repository and you complicate the work of the reviewer.

Vyzer9 commented:
Okay, thanks for the feedback. I won't be adding that branch name in future contributions. Do you think anything else is needed in the ReadMe?

Commit:
Merge branch 'master' into master

MarkEWaite reviewed (approved):
Looks good

MarkEWaite commented:
> Okay, thanks for the feedback. I won't be adding that branch name in future contributions. Do you think anything else is needed in the ReadMe?

I didn't see anything missing in my first review, but I'll leave the pull request open to wait for a second review.

Vyzer9 commented on README.md:
```diff
@@ -2,15 +2,32 @@
 
+---
+
+# 📌 Table of Contents
```
I'll leave it to the reader's eyes, it makes it easier and provides more clarity, but I'll remove it as you asked.

Commit:
Update README.md

Vyzer9 commented:
@lemeurherve I have already applied the changes you requested, thank you, I think it is now ok for the merge

MarkEWaite committed:
Merge branch 'master' into master

lemeurherve commented:
@Vyzer9 can you keep conversations unresolved until there is a change or acknowlegment please? (ex: https://github.com/jenkinsci/jenkins/pull/11147#discussion_r2403576112)

It makes it easier to follow for everyone.

Vyzer9 commented:
@lemeurherve Ok, I've already left it as unresolved, the other brother reviewed and approved the others.

Commit:
Merge branch 'master' into master

Vyzer9 commented:
@lemeurherve I left it there as unresolved but I have already made all the requested changes, I will be waiting for the response and possible merge

A1exKH reviewed (approved):
LGTM.

krisstern committed:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

MarkEWaite committed:
Merge branch 'master' into master

MarkEWaite reviewed (commented):
Issue tracker is moving to GitHub issues on Monday.  I asked about the 'good first issue' label in a post to the [developer list](https://groups.google.com/g/jenkinsci-dev/c/SEQ7UtbfRQ4/m/1O8sMMWjBAAJ).

MarkEWaite reviewed (changes requested):
URL for good first issues needs to be updated now that we've switched Jenkins core issues from Jira to GitHub.

MarkEWaite committed:
Update good first issues query

MarkEWaite committed:
Merge branch 'master' into master

MarkEWaite reviewed (approved):
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge
